src/App.tsx

```diff
@@ -1,26 +1,194 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useRef, useEffect, useState } from 'react';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import {
+  addTodo,
+  getTodos,
+  deleteTodo,
+  USER_ID,
+  updateTodo,
+} from './api/todos';
+import { Errors } from './types/Errors';
+import { Footer } from './components/Footer';
+import { ErrorModal } from './components/ErrorModal';
+import { FilterBy } from './types/FilterBy';
 
-const USER_ID = 0;
+const filter = (todos: Todo[], filterBy: FilterBy) => {
+  switch (filterBy) {
+    case FilterBy.Active:
+      return todos.filter(todo => !todo.completed);
+    case FilterBy.Completed:
+      return todos.filter(todo => todo.completed);
+    case FilterBy.All:
+    default:
+      return todos;
+  }
+};
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(Errors.DEFAULT);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadedTodoIds, setLoadedTodoIds] = useState<number[]>([]);
+
+  const shouldFocusCreationForm = useRef(false);
+
+  const withLoading = async (
+    todoId: number,
+    asyncCallback: () => Promise<void>,
+    options: { focusAfter?: boolean } = {},
+  ) => {
+    setLoadedTodoIds(current => [...current, todoId]);
+
+    try {
+      await asyncCallback();
+    } catch (error) {
```

Comment:
let's make it reusable , now it's look complicate you catch error here and than in every handler once more
also if your fucntion has more than two arguments it's should be object
```suggestion
const withLoading = async ({
  todoId,
  asyncCallback,
  options = {},
  errorMessage,
}: {
  todoId: number;
  asyncCallback: () => Promise<void>;
  options?: { focusAfter?: boolean };
  errorMessage: string
}) => {
    setLoadedTodoIds(current => [...current, todoId]);

    try {
      await asyncCallback();
    } catch (error) {
    setErrorMessage(errorMessage)
    ....
```